src/library_async.js

```diff
@@ -570,10 +586,30 @@ mergeInto(LibraryManager.library, {
     // exited, so that we know where the call stack began,
     // which is where we must call to rewind it.
     exportCallStack: [],
+    callStackNameToId: {},
+    callStackIdToFunc: {},
+#if ASYNCIFY_LAZY_LOAD_CODE
+    callStackIdToName: {},
+#endif
+    callStackId: 0,
     afterUnwind: null,
     asyncFinalizers: [], // functions to run when *all* asynchronicity is done
     sleepCallbacks: [], // functions to call every time we sleep
 
+    getCallStackId: function(funcName) {
+      var id = Asyncify.callStackNameToId[funcName];
+      if (id === undefined) {
+        id = Asyncify.callStackId++;
+        Asyncify.callStackNameToId[funcName] = id;
+#if ASYNCIFY_LAZY_LOAD_CODE
+        Asyncify.callStackIdToName[id] = funcName;
+#else
```

Comment:
the `else` here worries me - does this limit code using lazy-loading from also using asyncify normally? I'd expect lazy loading to add more, and not replace, in other words, but maybe I'm not understanding this yet.

(If this is an optimization, then I think it might be better to keep things simple and use the same method when lazy loading is enabled; in particular `getDataRewindFunc` would be simpler.)

Reply:
This is just the late-binding that I was talking about in my previous comment. Without lazy-loading we can grab the reference to the actual function immediately, but with lazy-loading we must resolve it later when we're about to call the function, because the `Module['asm']` will have been replaced by then.

If you'd like less conditional compilation here, I could just make it late-bind in all cases. This will add slight first-time-call overhead in the non-lazy case, but `#if ASYNCIFY_LAZY_LOAD_CODE` won't be needed at all.

Reply:
Oh, yes, now I see, thanks. Looks good.